cpp/src/arrow/filesystem/filesystem.h

```diff
@@ -115,6 +115,7 @@ struct ARROW_EXPORT FileStats {
 };
 
 /// \brief EXPERIMENTAL: file selector
+// call it FileSelector?
```

Comment:
`Selector` is a bit generic, shouldn't we rename it?

Reply:
Not in this PR, but I agree with `FileSelector` :-)

Reply:
Create JIRA https://issues.apache.org/jira/browse/ARROW-6463